### DISCUSSION TITLE
Single Plausible instance for tracking multiple domains/subdomains not working

### DISCUSSION THREAD

**Discussion by nemanjam** [plausible/analytics#4637] [Self-Hosted Support] [answered]
Like stated in docs, I should be able to track any website that has JavaScript snippet included regardless of the domain or subdomain where they are hosted, and later in dashboard I can filter by each domain.

https://plausible.io/docs/subdomain-hostname-filter

Similar discussion that says the same:

https://github.com/plausible/analytics/discussions/482

So I did exactly this, I am hosting 3 mirrors of same website that all have the same JavaScript snippet included.

1. https://nemanjamitic.com 

https://github.com/nemanjam/nemanjam.github.io/blob/main/.github/workflows/bash__deploy-nginx.yml

2. https://nemanjam.github.io

https://github.com/nemanjam/nemanjam.github.io/blob/main/.github/workflows/gh-pages__deploy-manual.yml

3. https://nmc-docker.arm1.nemanjamitic.com

https://github.com/nemanjam/nemanjam.github.io/blob/main/.github/workflows/default__build-push-docker.yml

But only the original domain https://nemanjamitic.com **that I set up in Plausible initial wizard is actually being tracked. Why?**

![image](https://github.com/user-attachments/assets/cbc663c0-7849-497e-b079-0c7eb739d702)

Here you can access the dashboard and see that only the original domain is tracked and there is no option to filter by domain:

https://plausible.arm1.nemanjamitic.com/nemanjamitic.com

Here is my `docker-compose.yml` for self hosting:

https://github.com/nemanjam/traefik-proxy/blob/main/apps/plausible/docker-compose.yml

How can I track all 3 of them, am I missing some settings, maybe this?

![image](https://github.com/user-attachments/assets/bb01cd00-55fc-4b66-9f57-f78507ff7b2a)

**Comment by ruslandoga** [comment, marked as answer]
👋 @nemanjam 

I'm not on the team / not very familiar with that part of Plausible, but I think the data-domain needs to be `nemanjamitic.com` for all sites. Right now it is `nemanjam.github.io` for the GitHub pages site and the request the tracker sends to Plausible doesn't provide any information on how to associate it with `nemanjamitic.com`

```sh
 curl 'https://plausible.arm1.nemanjamitic.com/api/event' \
  -X 'POST' \
  -H 'Content-Type: text/plain' \
  -H 'Host: plausible.arm1.nemanjamitic.com' \
  -H 'Origin: https://nemanjam.github.io' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.6 Safari/605.1.15' \
  -H 'Content-Length: 84' \
  --data-binary '{"n":"pageview","u":"https://nemanjam.github.io/","d":"nemanjam.github.io","r":null}'
```

So as far as Plausible is concerned, these are two separate sites.

**Reply by nemanjam** [reply]
What sense would it make, how can I then filter by domain if I send the same domain for all 3 mirrors?

You can see in this dscussion that he is explicitly setting different current domain with JavaScript and it's working for him.

https://github.com/plausible/analytics/discussions/482

```html
<script>
var script = document.createElement('script')
script.async=true
script.defer=true
script.dataset.domain=window.location.host // Set script data-domain dynamically
script.src = "https://plausible.io/js/plausible.js"
document.head.append(script)
</script>
```

**Reply by ruslandoga** [reply]
You would be able to filter by hostname because `event.hostname` is extracted from `u` or `url` which would stay https://nemanjam.github.io/

**Reply by ruslandoga** [reply]
https://github.com/plausible/analytics/discussions/482 is too old to be of any value :)

**Reply by nemanjam** [reply]
Ok, I can hardcode it easily, but why is he setting it explicitly differently? 

```html
script.dataset.domain=window.location.host 
```

Should I do some domain whitelisting in Plausible dashboard?

**Reply by ruslandoga** [reply]
> Ok, I can hardcode it easily, but why is he setting it explicitly differently?

Times were different :)

**Reply by nemanjam** [reply]
It looks like you are right, I can filter by `nemanjam.github.io` domain after I hardcoded `nemanjamitic.com` for all mirrors.

![image](https://github.com/user-attachments/assets/7b9f4108-5793-4f04-9aba-58142576bf9d)
